platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/internal/DefaultJavaFeatureSpec.java

```diff
@@ -58,6 +60,12 @@ public void capability(String group, String name, String version) {
 
     @Override
     public void create() {
+        NamedDomainObjectSet<JvmSoftwareComponentInternal> jvmComponents = project.getComponents().withType(JvmSoftwareComponentInternal.class);
+        if (jvmComponents.size() > 1) {
+            String componentNames = CollectionUtils.join(", ", jvmComponents.getNames());
+            throw new GradleException("Can not register feature '" + name + "'.  Can not register features using the java extension if multiple jvm components are present.  These components were found: '" + componentNames + "'.");
```

Comment:
```suggestion
            throw new GradleException("Cannot register feature '" + name + "' because multiple JVM components are present.  These components were found: " + componentNames + ".");
```